metadata-ingestion/src/datahub/api/entities/assertion/field_assertion.py

```diff
@@ -30,9 +29,11 @@
 )
 
 
-class FieldValuesFailThreshold(v1_ConfigModel):
-    type: Literal["count", "percentage"] = v1_Field(default="count")
-    value: int = v1_Field(default=0)
+class FieldValuesFailThreshold(BaseModel):
+    model_config = {"extra": "forbid"}
+
+    type: Literal["count", "percentage"] = Field(default="count")
+    value: int = Field(default=0)
```

Comment:
I know it is not part of this pr, but at line 42 the comparison is not correct -> 
It is comparing self.type (which is a string at runtime) to Literal["count"] (which is a type annotation, not a value). This comparison will always be False.